### PR TITLE
feat(VIZ-470): disable range selections in direct query

### PR DESCRIPTION
**Pull request by LiKang6688**

### CODE REVIEW

File: src/hooks/use-models.js [hunk 1]
```diff
@@ -44,7 +44,11 @@ const useModels = ({ core, flags }) => {
   const rect = useRect();
   const selections = useSelections();
   const app = useApp();
-  const { qLocaleInfo: localeInfo, qIsDirectQueryMode, qUnsupportedFeature } = useAppLayout();
+  const {
+    qLocaleInfo: localeInfo,
+    qIsDirectQueryMode: isDirectQueryMode,
+    qUnsupportedFeature: unsupportedFeature,
```
Comment (cbt1) [inline src/hooks/use-models.js]: Why are you renaming the properties?

File: src/hooks/use-models.js [hunk 2]
```diff
@@ -196,6 +200,8 @@ const useModels = ({ core, flags }) => {
       tooltipService,
       propertiesModel,
       trendLinesService,
+      isDirectQueryMode,
```
Comment (cbt1) [inline src/hooks/use-models.js]: These are not services or models and should not be exposed through the models hook. You are better off exposing them as a property on the layout service meta. As that is already being passed to the selectables function.

File: src/hooks/use-models.js (hunk 1 above)
Comment (LiKang6688) [reply src/hooks/use-models.js]: I followed the pattern - rename qLocaleInfo to localeInfo to rename qIsDirectQueryMode and qUnsupportedFeature

File: src/hooks/use-models.js (hunk 2 above)
Comment (LiKang6688) [reply src/hooks/use-models.js]: Good point. I will do that.

File: src/hooks/use-models.js (hunk 1 above)
Comment (cbt1) [reply src/hooks/use-models.js]: IMO keeping the name gives more value than changing it, as it kind of hints that the properties are internal and not actually coming from the layout. Which is not the case here. But if you where to mutade them in some way, sure then it makes sense to name them something else.

So from my point of view localeInfo should also change back to its original name.. But you can ignore that in this PR.

Comment (LiKang6688) [reply src/hooks/use-models.js]: I see. Thanks. Let me revert the commit.

File: src/picasso-definition/selectables/range/index.js [hunk 3]
```diff
@@ -17,39 +18,43 @@ export default function createRange({ models, actions, scales, chart }) {
...
+    ? createBinYRange({
+        actions,
+        selectionService,
+        dockService,
+        chart,
+        enableInteraction: () => dataHandler.getMeta().isBinnedData,
+      }) || EMPTY
+    : EMPTY;
 
   const legendRange =
```
Comment (cbt1) [inline src/picasso-definition/selectables/range/index.js]: Could this not be simplified to a single if statement? It does look somewhat more complex than needed with both a ternary operator and a logical operator, in multiple places as well.

```js
// Something like this on line 20-21, else return everything as before
if (!isRangeSelectionsSupported) {
  return createLegend({
      actions,
      selectionService,
      scales,
      legend,
      enableInteraction: () => true,
    }) || EMPTY;
}
```

Comment (LiKang6688) [reply src/picasso-definition/selectables/range/index.js]: Sure. Done

**Review by cbt1 — APPROVED** [review]

File: src/picasso-definition/selectables/range/index.js [hunk 4]
```diff
@@ -60,6 +61,13 @@ export default function createRange({ models, actions, scales, chart }) {
       enableInteraction: () => true,
     }) || EMPTY;
 
+  if (!isRangeSelectionsSupported) {
```
Comment (cbt1) [inline src/picasso-definition/selectables/range/index.js]: Consider putting this before creating all other range definitions

Comment (cbt1) [reply src/picasso-definition/selectables/range/index.js]: There is just a cost associated with creating those definitions and then end up not using them. So I see no value added by doing so.

Comment (LiKang6688) [reply src/picasso-definition/selectables/range/index.js]: Right, fixed.

**Pull request closed without merging**